src/framework/theme/components/button/base-button.directive.ts

```diff
@@ -0,0 +1,200 @@
+
+  /**
+   * Sets `outline` appearance
+   */
+  @Input()
+  @HostBinding('class.appearance-outline')
+  get outline(): boolean {
+    return this.appearance === 'outline';
+  }
+
+  set outline(value: boolean) {
+    if (convertToBoolProperty(value)) {
+      this.appearance = 'outline';
+    }
+  }
+
+  static ngAcceptInputType_outline: NbBooleanInput;
+
+  /**
+   * Sets `ghost` appearance
+   */
+  @Input()
+  @HostBinding('class.appearance-ghost')
+  get ghost(): boolean {
+
+    return this.appearance === 'ghost';
+  }
+
+  set ghost(value: boolean) {
+    if (convertToBoolProperty(value)) {
+      this.appearance = 'ghost';
+    }
+  }
+
+  static ngAcceptInputType_ghost: NbBooleanInput;
+
+  /**
+   * If set element will fill its container
+   */
+  @Input()
+  @HostBinding('class.full-width')
+  get fullWidth(): boolean {
+    return this._fullWidth;
+  }
+
+  set fullWidth(value: boolean) {
+    this._fullWidth = convertToBoolProperty(value);
+  }
+
+  private _fullWidth = false;
+  static ngAcceptInputType_fullWidth: NbBooleanInput;
+
+  /**
+   * Disables the button
+   */
+  @Input()
+  @HostBinding('attr.aria-disabled')
+  @HostBinding('class.btn-disabled')
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value: boolean) {
+    this._disabled = convertToBoolProperty(value);
+    this.renderer.setProperty(this.hostElement.nativeElement, 'disabled', this.disabled);
+  }
+
+  private _disabled: boolean = false;
+  static ngAcceptInputType_disabled: NbBooleanInput;
+
+  @HostBinding('class.size-tiny')
+  get tiny() {
+    return this.size === 'tiny';
+  }
+
+  @HostBinding('class.size-small')
+  get small() {
+    return this.size === 'small';
+  }
+
+  @HostBinding('class.size-medium')
+  get medium() {
+    return this.size === 'medium';
+  }
+
+  @HostBinding('class.size-large')
+  get large() {
+    return this.size === 'large';
+  }
+
+  @HostBinding('class.size-giant')
+  get giant() {
+    return this.size === 'giant';
+  }
+
+  @HostBinding('class.shape-rectangle')
+  get rectangle() {
+    return this.shape === 'rectangle';
+  }
+
+  @HostBinding('class.shape-round')
+  get round() {
+    return this.shape === 'round';
+  }
+
+  @HostBinding('class.shape-semi-round')
+  get semiRound() {
+    return this.shape === 'semi-round';
+  }
+
+  @HostBinding('class.icon-start')
+  get iconLeft(): boolean {
+    const el = this.hostElement.nativeElement;
+    const icon = this.iconElement;
+    return !!(icon && firstChildNotComment(el) === icon);
+  }
+
+  @HostBinding('class.icon-end')
+  get iconRight(): boolean {
+    const el = this.hostElement.nativeElement;
+    const icon = this.iconElement;
+    return !!(icon && lastChildNotComment(el) === icon);
+  }
+
+  protected constructor(
+    protected renderer: Renderer2,
+    protected hostElement: ElementRef<HTMLElement>,
+    // @breaking-change @7.0.0  make cd required
+    @Optional() protected cd?: ChangeDetectorRef,
```

Comment:
Remove comment and make `cd` required. Then remove `undefined` check on line 191 (`updateSmth`) method.